project/CRUD/roles.php

```diff
@@ -1,3 +1,173 @@
+    // by adding (array_push()) corresponding error unto $errors array
+    if (empty($role_name)) {
+        array_push($errors, "Role Name is required");
+    }
+
+    if (count($errors) == 0) {
+        $update = "UPDATE roles set role_name = '$role_name' WHERE role_id ='$id'";
+
+        if (mysqli_query($conn, $update)) {
+            array_push($success, "Update Successful");
+            // clear variables
+            $role_name = "";
+        } else {
+            array_push($errors, "Error updating role: ", mysqli_error($conn));
+        }
+    }
+}
+
+// DELETE
+if (isset($_GET['delete_id'])) {
+    $id = mysqli_real_escape_string($conn, $_GET['delete_id']);
+    $delete = "DELETE FROM roles WHERE role_id='$id'";
+    if (mysqli_query($conn, $delete)) {
+        array_push($success, "Delete successful");
+    } else {
+        array_push($errors, "Delete error: " . mysqli_error($conn));
+    }
+}
+
+?>
+
+<div class="content-body">
+    <?php if (isset($_GET['delete_view'])) {
+        display_success();
+        display_error();
+    }
+
+    $query = "SELECT * FROM roles ORDER BY role_id ASC";
+    $results = mysqli_query($conn, $query);
+
+    ?>
+    <p><b>Roles</b></p>
+    <table>
+        <thead>
+            <tr>
+                <?php isAdmin() ? print '<th>Role ID</th>' : ''; ?>
+                <th>Role Name</th>
+                <?php isAdmin() ? print '<th colspan="2">Action</th>' : ''; ?>
+
+            </tr>
+        </thead>
+        <tbody>
+            <?php
+            while ($row = mysqli_fetch_assoc($results)) {
+                $id = $row['role_id'];
+                $role_name = $row['role_name'];
+            ?>
+                <tr>
+                    <?php if (isAdmin()) {
+                        echo '<td>' . $id . '</td>';
+                    } ?>
+                    <td><?php echo $role_name ?></td>
+                    <?php if (isAdmin()) {
+                        echo '<td><a href="?page=roles&update_view=true&update_id=' . $id . '">Update</a></td>';
+                        echo '<td><a href="?page=roles&delete_view=true&delete_id=' . $id . '">Delete</a></td>';
+                    } ?>
+                </tr>
+            <?php
+            }
+            ?>
+        </tbody>
+    </table>
+
+    <?php if (isAdmin()) { ?>
+        <a href="?page=roles&add_view=true">
+            <button>Add Role</button>
+        </a>
+    <?php } ?>
+
+    <?php if (isset($_GET['add_view'])) { ?>
+        <hr>
+        <div class="form-container">
+            <form class="form-body" action="" method="post">
+                <?php echo display_success(); ?>
+                <?php echo display_error(); ?>
+                <div class="form-input">
+                    <p><b>Add a role</b></p>
+                    <label>Role Name</label>
+                    <span><input type="text" name="role_name"></span>
+                </div>
+                <div class="form-submit">
```

Comment:
As per our DB, role table has 'role description' - so add role needs to have option to be able to add role description as well (just another input field - with respective name)